meta-mender-core/classes/mender-artifactimg.bbclass

```diff
@@ -55,6 +63,26 @@ IMAGE_CMD_mender () {
         image_flag=-f
     fi
 
+    if [ -n "${ARTIFACT_NAME_PROVIDES}" ]; then
+        extra_args="$extra_args  --provides ${ARTIFACT_NAME_PROVIDES}"
+    fi
+
+    if [ -n "${ARTIFACT_PROVIDES}" ]; then
+        extra_args="$extra_args  --provides ${ARTIFACT_PROVIDES}"
+    fi
+
+    if [ -n "${ARTIFACT_PROVIDES_GROUP}" ]; then
+        extra_args="$extra_args --provides-group ${ARTIFACT_PROVIDES_GROUP}"
+    fi
+
+    if [ -n "${ARTIFACT_DEPENDS}" ]; then
+        extra_args="$extra_args --depends ${ARTIFACT_DEPENDS}"
+    fi
+
+    if [ -n "${ARTIFACT_DEPENDS_GROUPS}" ]; then
+        extra_args="$extra_args --depends-groups ${ARTIFACT_DEPENDS_GROUPS}"
```

Comment:
If there is more than one entry, don't you need to loop over these and do some joining with `--depends-groups` in between (same for the others)?

Reply:
I guess it depends on how we require the input to be formed, right?
It should be fine with a string if the input is formatted like MA expects, I think..

Reply:
It doesn't work with `--depends-groups group1 group2`, only with `--depends-groups group1 --depends-groups group2`. It's quite buggy in that respect actually, because it *silently* accepts the incorrect command, and then uses only the first element.

Reply:
haha, lol, Yeh, I discovered it during testing :p

Reply:
https://tracker.mender.io/browse/MEN-2849